# PVCFPipe
Population VCF Pipeline: Streamline Population Analyses with VCF-Tools

Please Visit www.endemicbio.info/pvcfpipe for detailed documentation of pipeline and tutorials

PVCF-Pipe streamlines the processing, calculation, and output of population genetics metrics (i.e. Pi, Tajima's D, FIS, Observed/Expected Heterozygosity and Homozygosity) generated from VCF-Tools. This pipeline was generated due to a lack of tools for effectively processing VCF-Tools output and compiling all data for populations into a single csv file for analysis. The pipeline gets averages across all loci and sites for each metric, using the unfiltered input VCF to calculate Pi (assuming you want to keep all SNPs to calculate diversity) and using a filtered VCF for all other metrics based on minor allele frequency and SNP distance filtering parameters input by the user. VCF-Tools calculates Hardy-Weinberg equilibrium metrics for different loci, but the output is not properly formatted to calculate inbreeding co-effiecents (FIS = Hexp-Hobs/Hexp). PVCF-Pipe converts raw allele counts into proportions which are used to calculate FIS for each locus and are then averaged across all loci for the final output value. PVCF-Pipe offers users VCF filtering options (i.e. minor allele frequency, SNP linkage) and window sizes for Tajima's D. Finally, individual level metrics (i.e. using the VCF-Tools --het option) are generated and compiled into a single csv file that is annotated with population names for each sample. 
